### PR TITLE
`Integration Tests`: run on iOS 17

### PR DESCRIPTION
**Pull request by NachoSoto**
There is a lot of flaky failures due to bugs that have been fixed in iOS 17.

Depends on #3403, blocked by `FB13358220`.

### CODE REVIEW

**Comment by NachoSoto** [comment]
We're gonna have to figure out why this isn't working:

> Too long with no output (exceeded 20m0s): context deadline exceeded

**Comment by NachoSoto** [comment]
Seems like tests run reliably now!

**Review by tonidero — COMMENTED** [review]
Is this ready @NachoSoto ? Changes look good to me

**Comment by NachoSoto** [comment]
Yeah but it's blocked by https://discuss.circleci.com/t/severe-performance-problems-with-xcode-15/49205/11
I'm hoping the next beta will make it better.
If anybody wants to approve I'll merge when they actually pass 😇

**Review by tonidero — APPROVED** [review]

**Comment by MarkVillacampa** [comment]
Rebased main to see if tests finally pass on iOS 17

**Comment by NachoSoto** [comment]
Thanks for checking. Looks like they're even more flaky :/

File: .circleci/config.yml
Comment (NachoSoto) [inline .circleci/config.yml:1158]: REMOVE

File: .circleci/config.yml
Comment (MarkVillacampa) [inline .circleci/config.yml:724]: Do we want this?

Comment (NachoSoto) [reply .circleci/config.yml:724]: CircleCI is getting rid of Intel in June. So good to migrate these early.

**Pull request merged**
